pkg/util/azureclient/fake/compute.go

```diff
@@ -0,0 +1,206 @@
+	for _, vm := range s.az.Vms {
+		if *ss.Name == *vm.Tags["scaleset"] {
+			have++
+		}
+	}
+	s.az.log.Debugf("scale have:%d, cap:%d", have, *ss.Sku.Capacity)
+	if have > *ss.Sku.Capacity {
+		return fmt.Errorf("should not be automatically scaling down")
+	}
+	for v := have; *ss.Sku.Capacity > have; v++ {
+		name := fmt.Sprintf("%s-%d", (*ss.Name)[3:], v)
+		compName := fmt.Sprintf("%s-%06d", (*ss.Name)[3:], v)
+		s.az.log.Infof("scale have:%d, cap:%d, v:%d, name:%s, compName:%s", have, *ss.Sku.Capacity, v, name, compName)
+		tags := ss.Tags
+		if tags == nil {
+			tags = map[string]*string{}
+		}
+		tags["scaleset"] = ss.Name
+		vm := compute.VirtualMachineScaleSetVM{
+			ID:         to.StringPtr(uuid.NewV4().String()),
+			InstanceID: to.StringPtr(uuid.NewV4().String()),
+			Name:       &name,
+			VirtualMachineScaleSetVMProperties: &compute.VirtualMachineScaleSetVMProperties{
+				OsProfile: &compute.OSProfile{ComputerName: &compName},
+			},
+			Location: ss.Location,
+			Plan:     ss.Plan,
+			Tags:     tags,
+			Zones:    ss.Zones,
+		}
+		s.az.Vms = append(s.az.Vms, vm)
+		s.az.VirtualMachineScaleSetVMsClient.Start(ctx, resourceGroupName, *ss.Name, *vm.InstanceID)
+		have++
+	}
+	return nil
+}
+
+// CreateOrUpdate Fakes base method
+func (s *FakeVirtualMachineScaleSetsClient) CreateOrUpdate(ctx context.Context, resourceGroupName, VMScaleSetName string, parameters compute.VirtualMachineScaleSet) error {
+	found := false
+	for i, ss := range s.az.Ssc {
+		if VMScaleSetName == *ss.Name {
+			found = true
+			s.az.Ssc[i] = parameters
+			break
+		}
+	}
+	if !found {
+		s.az.Ssc = append(s.az.Ssc, parameters)
+	}
+	s.scale(ctx, resourceGroupName, &parameters)
+	return nil
+}
+
+// Delete Fakes base method
+func (s *FakeVirtualMachineScaleSetsClient) Delete(ctx context.Context, resourceGroupName, VMScaleSetName string) error {
+	for i, ss := range s.az.Ssc {
+		if VMScaleSetName == *ss.Name {
+			*ss.Sku.Capacity = 0
+			s.scale(ctx, resourceGroupName, &ss)
+			s.az.Ssc = append(s.az.Ssc[:i], s.az.Ssc[i+1:]...)
+			return nil
+		}
+	}
+	return nil
+}
+
+// List Fakes base method
+func (s *FakeVirtualMachineScaleSetsClient) List(ctx context.Context, resourceGroup string) ([]compute.VirtualMachineScaleSet, error) {
+	return s.az.Ssc, nil
+}
+
+// Update Fakes base method
+func (s *FakeVirtualMachineScaleSetsClient) Update(ctx context.Context, resourceGroupName, VMScaleSetName string, parameters compute.VirtualMachineScaleSetUpdate) error {
+	for _, ss := range s.az.Ssc {
+		if VMScaleSetName == *ss.Name {
+			// the scaler changes the capacity
+			ss.Sku.Capacity = parameters.Sku.Capacity
+			s.scale(ctx, resourceGroupName, &ss)
+			return nil
+		}
+	}
+	return nil
+}
+
+// UpdateInstances Fakes base method
+func (s *FakeVirtualMachineScaleSetsClient) UpdateInstances(ctx context.Context, resourceGroupName, VMScaleSetName string, VMInstanceIDs compute.VirtualMachineScaleSetVMInstanceRequiredIDs) error {
+	// not implemented yet
+	return nil
```

Comment:
don't know if this should return a not implemented error

Reply:
good idea